src/ansiblelint/_internal/rules.py

```diff
@@ -104,6 +104,28 @@ def matchtask(
         the original task (before normalization) will be made available under
         ``task["__raw_task__"]``.
         """
+
```

Comment:
You do not want to modify implementation matchtask, instead you want to modify the implementation of `iter_tasks_in_file` function, which is called by matchtask (the function calling matchtask for each task).

Basically adding the inherited tasks should be done automatically by `iter_tasks_in_file`.